### PR TITLE
Replace MetricsManager with MetricRegistry

### PR DESCRIPTION
**Pull request by gerolf-da**
Instead of passing around MetricsManager just for the timedFuture
method, we pass the underlying MetricRegistry directly and use
the timedFuture function as an actual function (passing in a timer metric).

The Sandbox still starts the JmxReporter, but the integrator of the Indexer
and the Ledger API Server needs to enable the reporting facilities themselves.

Contributes to #3513.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala [hunk 1]
```diff
@@ -61,13 +62,18 @@ object ReferenceServer extends App {
...
+      participantLoggerFactory,
+      SharedMetricRegistries.getOrCreate(s"indexer-$participantId"))
     indexServer <- StandaloneIndexServer(
       config,
       readService,
       writeService,
       authService,
-      participantLoggerFactory).start()
+      participantLoggerFactory,
+      SharedMetricRegistries.getOrCreate(s"ledger-api-server-$participantId")).start()
```
Comment (SamirTalwar) [inline ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: Can I recommend we start adding a trailing `,` and newline to multi-line function calls to make the diffs cleaner?

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala [hunk 2]
```diff
@@ -61,13 +62,18 @@ object ReferenceServer extends App {
 
   val participantLoggerFactory = NamedLoggerFactory.forParticipant(participantId)
   val participantF: Future[(AutoCloseable, StandaloneIndexServer#SandboxState)] = for {
-    indexerServer <- StandaloneIndexerServer(readService, config, participantLoggerFactory)
+    indexerServer <- StandaloneIndexerServer(
+      readService,
+      config,
+      participantLoggerFactory,
+      SharedMetricRegistries.getOrCreate(s"indexer-$participantId"))
```
Comment (SamirTalwar) [inline ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: Perhaps pull the registry out (and `SharedMetricRegistries.getOrCreate(s"ledger-api-server-$participantId")` too), as they're used twice.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala [hunk 3]
```diff
@@ -35,25 +31,16 @@ final class MetricsManager(jmxDomain: String, enableJmxReporter: Boolean) extend
...
 
-object MetricsManager {
-  def apply(jmxDomain: String, enableJmxReporter: Boolean = true): MetricsManager =
-    new MetricsManager(jmxDomain, enableJmxReporter)
+object MetricsReporting {
+  def apply(
+      metrics: MetricRegistry,
+      jmxDomain: String,
+      enableJmxReporter: Boolean = true): MetricsReporting =
+    new MetricsReporting(metrics, jmxDomain, enableJmxReporter)
```
Comment (SamirTalwar) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: Why bother with this? We can just call the constructor directly.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala [hunk 4]
```diff
@@ -35,25 +31,16 @@ final class MetricsManager(jmxDomain: String, enableJmxReporter: Boolean) extend
...
 }
 
-object MetricsManager {
-  def apply(jmxDomain: String, enableJmxReporter: Boolean = true): MetricsManager =
-    new MetricsManager(jmxDomain, enableJmxReporter)
+object MetricsReporting {
+  def apply(
+      metrics: MetricRegistry,
+      jmxDomain: String,
+      enableJmxReporter: Boolean = true): MetricsReporting =
```
Comment (SamirTalwar) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: We never set `enableJmxReporter` to `false`. Can we just always do it?

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: yeah, good idea. I'm not yet consistent with doing so. Thanks for the reminder.

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 2 above)
Comment (gerolf-da) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: `participantId` refers to two different IDs.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala (hunk 3 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: Consider it gone.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala (hunk 4 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: Yes, I think it is safe to do so now.

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 2 above)
Comment (SamirTalwar) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: Good point. Can we avoid the shadowing by renaming the other one to `extraParticipantId` or something?

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 1 above)
Comment (SamirTalwar) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: I think later versions of Scalafmt can do this for us.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala (hunk 4 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: Addressed by 1227da2.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala (hunk 3 above)
Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporting.scala]: Addressed by 1227da2.

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 2 above)
Comment (gerolf-da) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: Addressed by 1227da2.

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: Addressed by 1227da2.

**Pull request merged**